Sources/LiveViewNativeAVKit/VideoPlayerView.swift

```diff
@@ -0,0 +1,215 @@
+//
+//  AVKitRegistry.swift
+//  LiveViewNativeAVKit
+//
+//  Created by May Matyi on 5/18/23.
+//
+
+import SwiftUI
+import AVFoundation
+import AVKit
+import LiveViewNative
+import CoreMedia
+
+/// A native video player view. It can be rendered in a LiveViewNative app using the `VideoPlayer` element.
+/// 
+/// ```elixir
+/// defmodule MyApp.VideoPlayerLive do
+///   use Phoenix.LiveView
+///   use LiveViewNative.LiveView
+/// 
+///   native_binding :is_muted, Atom, false
+///   native_binding :time_control_status, String, ""
+///   native_binding :playback_time, Float, 5.0
+///   native_binding :playback_time_update_interval, Float, 0.05
+/// 
+///   @impl true
+///   def render(%{platform_id: :swiftui} = assigns) do
+///     ~SWIFTUI"""
+///     <VStack id="webview-example">
+///       <VStack>
+///         <%= @playback_time %>
+///       </VStack>
+///       <VideoPlayer
+///         autoplay
+///         is-muted="is_muted"
+///         playback-time="playback_time"
+///         playback-time-update-interval="playback_time_update_interval"
+///         on-play="handle_play"
+///         on-pause="handle_pause"
+///         time-control-status="time_control_status"
+///         url="http://192.168.1.143:4000/videos/sample2.mp4"
+///         volume="volume" />
+///     </VStack>
+///     """
+///   end
+/// end
+/// ```
+///
+/// ## Attributes
+/// * ``autoplay``
+/// * ``url``
+///
+/// ## Bindings
+/// * ``isMuted``
+/// * ``playbackTime``
+/// * ``playbackTimeUpdateInterval``
+/// * ``timeControlStatus``
+///
+/// ## Events
+/// - ``pauseEvent``
+/// - ``playEvent``
+#if swift(>=5.8)
+@_documentation(visibility: public)
+#endif
+struct VideoPlayerView<R: RootRegistry>: View {
+    /// An event that is fired when the video player is paused.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    @Event("on-pause", type: "click") private var pause
+
+    /// An event that is fired when the video player is played.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    @Event("on-play", type: "click") private var play
+
+    /// A boolean indicating whether the video player is muted.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    @LiveBinding(attribute: "is-muted") private var isMuted: Bool
+
+    /// The current playback time of the video player in seconds.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    @LiveBinding(attribute: "playback-time") private var playbackTime: Float64
+
+    /// The interval at which the playback time is updated.
+    #if swift(>=5.8)
+    @_documentation(visibility: public)
+    #endif
+    @LiveBinding(attribute: "playback-time-update-interval") private var playbackTimeUpdateInterval: Float64
```

Comment:
This seems like it can just be an `@Attribute`, since it doesn't need to be synchronized. It could then be passed to the `VideoPlayerObserver` in `init` as the interval and avoid the extra `throttle`.

Reply:
How would I go about doing this? When I change `playbackTimeUpdateInterval` to an `Attribute` and try to pass it to `VideoPlayerObserver` in `init`, I get the error `Variable 'self.observer' used before being initialized`.

Reply:
You could instead use `element.attributeValue(for:)` in the `init`, since the element node is passed in there.